squeasel.c

```diff
@@ -3788,6 +3793,12 @@ static int set_uid_option(struct sq_context *ctx) {
 static pthread_mutex_t *ssl_mutexes;
 
 static int sslize(struct sq_connection *conn, SSL_CTX *s, int (*func)(SSL *)) {
+  char first_byte;
+  // TLS handshake always starts with a ContentType byte equal to 22 (RFC8446 5.1).
+  if (!recv(conn->client.sock, &first_byte, 1, MSG_PEEK) || first_byte != 22) {
+    return 0;
+  }
```

Comment:
What exactly are we trying to achieve with this? Is this a bugfix for HTTP connections to an HTTPS port? Can you clarify in the description?

Reply:
Yes, this is so we don't eat up bytes from the socket assuming it's HTTPS before realizing it's HTTP.